### PR TITLE
Schizophreni branch

### PR DESCRIPTION
**Pull request by Schizophreni**
Add code for setting the demand of GPU with tensorflow 2.0
Simplify the codes for softmax of numpy implementation in Chap4 tutorial_minst_fnn-numpy-exercise.ipynb, including forward and backward
Add a brief and simpler derivation of back propogation for softmax. 
All codes changed have been annotated with "Changed by Schizophreni", the author can utilize this to find the changes and modify the code.

### CODE REVIEW

**Comment by jingjing-gong** [comment]
梯度推导是对的，inline的latex用\$\$\<your latex\>\$\$，矢量应该用粗体表示 x、 y符号有点乱，建议不使用用矢量表示。另外不要开头的gpu配置，这是一个教程project，gpu的设置在这里是一个增加复杂度的操作。注释中不要加入 “by Schizophreni”，这对读者没啥帮助，你的contribution已经被github记录的很好了。“避免上溢和下溢”建议改成“避免在exp操作时候数值上溢”

**Comment by jingjing-gong** [comment]
建议最终的推导结果写成这样：
`$$\frac{\partial L}{\partial a_j} = \frac{\partial L}{\partial s_j} s_j - s_j\sum_{i}{\frac{\partial L}{\partial s_i}s_i}$$`

**Comment by Schizophreni** [comment]
谢谢回复。github确实可以记录下代码增删的记录，确实不应该加上“by Schizophreni"，公式的问题也存在。谢谢作者指出！
祝好！




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "JerrikEph"<notifications@github.com&gt;; 
发送时间: 2020年1月2日(星期四) 晚上6:25
收件人: "nndl/exercise"<exercise@noreply.github.com&gt;; 
抄送: "gone with the wind"<1371482301@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [nndl/exercise] Schizophreni branch (#15)




梯度推导是对的，inline的latex用$$<your latex&gt;$$，矢量尽量用粗体。另外不要开头的gpu配置，这是一个教程project，gpu的设置在这里是一个增加复杂度的操作。注释中不要加入 “by Schizophreni”，这对读者没啥帮助，你的contribution已经被github记录的很好了。“避免上溢和下溢”建议改成“避免在exp操作时候数值上溢”
